### ISSUE TITLE
[Feature Request] Allow adding certificate as Base64 String to ConfidentialClientApplication

### ISSUE THREAD

**Issue by LaurensAdema** [AzureAD/microsoft-authentication-library-for-dotnet#2886]
**Is your feature request related to a problem? Please describe.**
We load our certificate's from Azure Key Vault using the IConfiguration provider. This means that certificates are stored as Base64 strings in the provider. When building ConfidentialClient applications we need to convert these to X509Certificate2 which according to the profiler uses a lot of CPU capacity. Since we build quite a lot of these ConfidentialClient applications this would result in a significant performance boost.

**Describe the solution you'd like**
Adding a certificate to an Confidential Client application as Base64 string.

**Describe alternatives you've considered**
We've considered stepping away from certificate's altogether or building a sort of X509Certificate cache if you will, but this wasn't desirable considering the potential issues.

**Additional context**
Code sample of current (slow) solution:
```cs
X509Certificate2 certificate = new X509Certificate2(Convert.FromBase64String(Configuration["Application:Certificate"]));

IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(applicationId)
    .WithTenantId(tenantId.ToString())
    .WithCertificate(certificate)
    .WithLegacyCacheCompatibility(false)
    .Build();
```

**Comment by bgavrilMS** [comment]
Hi @LaurensAdema 

MSAL uses the certificate to create and sign a small piece of JSON (an assertion) which it then sends to AAD. It does not send the whole cert over. MSAL needs at least: 

- cert.GetCertHash() // we send the cert thumbprint in the assertion, so that AAD can match it with the cert it has
- cert.Sign(); // MSAL signs the assertion

When it receives the request, AAD first looks at the thumbpritn in the assertion. With it, it locates the certificate you have configured during your app registration (since you can have more than 1). This certificate only has a public key, which is used to verify the digital signature.

So the validation that your confidential client is actually secure, i.e. it is in possession of a cert is done based on digital signature, not by sending the actual secret, which would be insecure. 


So, to better understand your request - are you saying that you'd like MSAL to cache the certificate objects for you ? (e.g. based on thumbprint)? And maybe attempt to do some cert rotation for you?

Microsoft.identity.Web does some of this by the way, see https://github.com/AzureAD/microsoft-identity-web/wiki/Certificates#microsoft-identity-web-classes-used-for-certificate-management

**Comment by LaurensAdema** [comment]
Yes, that would be wonderful. We use the Microsoft.Identity.Web solution in the link you've provided where this is posible, but this is not yet possible on the ConfidentialClients right? I saw only ClientSecret could be loaded with CreateWithApplicationOptions.

**Comment by jmprieur** [comment]
@LaurensAdema : look at this sample:
https://github.com/Azure-Samples/active-directory-dotnetcore-daemon-v2/tree/master/1-Call-MSGraph/daemon-console

in particular: https://github.com/Azure-Samples/active-directory-dotnetcore-daemon-v2/blob/e9b986b14c3911a53a593b84f1d9e16dc3d9e950/1-Call-MSGraph/daemon-console/Program.cs#L140 where you can use the loading by thumbprint
you could also read directly a certificate description, and use the DefaultCertificateLoader class: https://github.com/AzureAD/microsoft-identity-web/wiki/asp-net#help-loading-certificates
